ocean_lib/data_provider/data_service_provider.py

```diff
@@ -103,7 +102,137 @@ def encrypt(
+        payload = {"did": did, "serviceId": service_id}
+
+        response = DataServiceProvider._http_method(
+            "post", service_endpoint, json=payload
+        )
+
+        if not response or not hasattr(response, "status_code"):
+            raise Exception("Response not found!")
+
+        if response.status_code != 200:
+            msg = (
+                f"Fileinfo service failed at the FileInfoEndpoint "
+                f"{service_endpoint}, reason {response.text}, status {response.status_code}"
+            )
+            logger.error(msg)
+            raise Exception(msg)
+
+        logger.info(
+            f"Retrieved asset files successfully"
+            f" FileInfoEndpoint {service_endpoint}"
+        )
+        return response
+
+    @staticmethod
+    @enforce_types
+    def initialize(
+        did: str,
+        service_id: str,
+        consumer_address: str,
+        service_endpoint: str,
+        userdata: Optional[Dict] = None,
+    ) -> Response:
+
+        req = PreparedRequest()
+        service_endpoint += f"?documentId={did}"
+        payload = {
+            "serviceId": service_id,
+            "consumerAddress": consumer_address,
+        }
+
+        if userdata:
+            userdata = json.dumps(userdata)
+            payload["userdata"] = userdata
+
+        req.prepare_url(service_endpoint, payload)
+
+        response = DataServiceProvider._http_method("get", req.url)
+
+        if not response or not hasattr(response, "status_code"):
+            raise Exception("Response not found!")
+
+        if response.status_code != 200:
+            msg = (
+                f"Initialize service failed at the initializeEndpoint "
+                f"{req.url}, reason {response.text}, status {response.status_code}"
+            )
+            logger.error(msg)
+            raise Exception(msg)
+
+        logger.info(
+            f"Service initialized successfully" f" initializeEndpoint {req.url}"
+        )
+
+        return response
+
+    @staticmethod
+    @enforce_types
+    def download(
+        did: str,
+        service_id: str,
+        tx_id: str,
+        consumer_wallet: Wallet,
+        service_endpoint: str,
+        destination_folder: Union[str, Path],
+        index: Optional[int] = None,
+        userdata: Optional[Dict] = None,
+    ) -> None:
+
+        fileinfo_response = DataServiceProvider.fileinfo(
+            did, service_id, "http://172.15.0.4:8030/api/services/fileinfo"
+        )
+        files = fileinfo_response.json()
+        indexes = range(len(files))
+        if index is not None:
+            assert isinstance(index, int), logger.error("index has to be an integer.")
+            assert index >= 0, logger.error("index has to be 0 or a positive integer.")
+            assert index < len(files), logger.error(
+                "index can not be bigger than the number of files"
+            )
+            indexes = [index]
```

Comment:
btw the correct plural of "index" is "indices"

Reply:
https://stackoverflow.com/questions/1378781/proper-terminology-should-i-say-indexes-or-indices both are correct :))